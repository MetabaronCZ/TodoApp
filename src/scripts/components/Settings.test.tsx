import React from 'react';
import { Provider } from 'react-redux';

import { act, render } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';

import { Settings } from 'components/Settings';
import { Dropdown } from 'components/forms/Dropdown';

import { mockStore } from 'test/store';
import { client } from 'modules/client';
import { perPages } from 'models/Settings';
import { withMockedProviders } from 'test/component';

jest.mock('components/forms/Dropdown');

const getSettings = (): JSX.Element => {
  const store = mockStore();
  return withMockedProviders(
    <Provider store={store}>
      <Settings />
    </Provider>,
  );
};

describe('components/Settings', () => {
  it('should render correctly', () => {
    const mockedDropdown = jest.mocked(Dropdown);
    expect(mockedDropdown).toBeCalledTimes(0);

    const { container } = render(getSettings());
    expect(container).toMatchSnapshot();
    expect(mockedDropdown).toBeCalledTimes(1);

    const props = mockedDropdown.mock.calls[0][0];
    expect(props.items.length).toEqual(perPages.length);
  });

  it('should auto-update settings when perPage changes', async () => {
    const testPerPage = 50;

    const mockedDropdown = jest.mocked(Dropdown);
    const mockedUpdateSettings = jest.spyOn(client.settings, 'set');
    expect(mockedDropdown).toBeCalledTimes(0);
    expect(mockedUpdateSettings).toBeCalledTimes(0);

    render(getSettings());
    expect(mockedDropdown).toBeCalledTimes(1);

    act(() => {
      const props = mockedDropdown.mock.calls[0][0];
      props.onSelect(testPerPage);
    });

    expect(mockedUpdateSettings).toBeCalledTimes(1);
    expect(mockedUpdateSettings).toBeCalledWith({ perPage: testPerPage });
  });
});
