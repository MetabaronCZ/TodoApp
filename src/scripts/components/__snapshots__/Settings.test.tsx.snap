// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Settings should render correctly 1`] = `
.c0 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 18px;
  line-height: 30px;
  font-weight: normal;
  margin: 0;
}

.c1 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  border-bottom: 1px solid #c0c0c0;
}

.c2:last-child {
  border-bottom: none;
}

.c5 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  color: #222;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c11 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

.c9 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-right: 5px;
  background: #f0f0f0;
}

.c9:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c9:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c9:disabled {
  color: #c0c0c0;
}

.c10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c7 {
  position: relative;
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 10px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  min-height: 30px;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div>
    <h1
      class="c0"
    >
      Nastavení
    </h1>
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="form-field-0"
            >
              Stránkování:
            </label>
          </div>
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <button
                class="c8 c9"
                id="form-field-0"
                type="button"
              >
                <span
                  class="c10"
                >
                  20
                </span>
                <i
                  class="c11"
                >
                  ˅
                </i>
              </button>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
